pulsar-broker/src/main/java/org/apache/pulsar/broker/service/BrokerService.java

```diff
@@ -403,15 +411,41 @@ public BrokerService(PulsarService pulsar, EventLoopGroup eventLoopGroup) throws
         this.defaultServerBootstrap = defaultServerBootstrap();
 
         this.pendingLookupRequests = ObserverGauge.build("pulsar_broker_lookup_pending_requests", "-")
-                .supplier(() -> pulsar.getConfig().getMaxConcurrentLookupRequest()
-                        - lookupRequestSemaphore.get().availablePermits())
+                .supplier(this::getPendingLookupRequest)
                 .register();
+        this.pendingLookupRequestsCounter = pulsar.getOpenTelemetry().getMeter()
+                .gaugeBuilder("pulsar.broker.lookup.pending.request.usage")
+                .ofLongs()
+                .setDescription("The number of pending lookup requests in the broker. "
+                        + "When it reaches threshold \"maxConcurrentLookupRequest\" defined in broker.conf, "
+                        + "new requests are rejected.")
+                .buildWithCallback(measurement -> measurement.record(getPendingLookupRequest()));
+        this.pendingLookupRequestsLimit = pulsar.getOpenTelemetry().getMeter()
+                .gaugeBuilder("pulsar.broker.lookup.pending.request.limit")
+                .ofLongs()
+                .setDescription("The maximum number of pending lookup requests in the broker. "
+                        + "Equal to \"maxConcurrentLookupRequest\" defined in broker.conf.")
+                .buildWithCallback(
+                        measurement -> measurement.record(pulsar.getConfig().getMaxConcurrentLookupRequest()));
 
         this.pendingTopicLoadRequests = ObserverGauge.build(
-                "pulsar_broker_topic_load_pending_requests", "-")
-                .supplier(() -> pulsar.getConfig().getMaxConcurrentTopicLoadRequest()
-                        - topicLoadRequestSemaphore.get().availablePermits())
+                        "pulsar_broker_topic_load_pending_requests", "-")
+                .supplier(this::getPendingTopicLoadRequests)
                 .register();
+        this.pendingTopicLoadRequestsCounter = pulsar.getOpenTelemetry().getMeter()
+                .gaugeBuilder("pulsar.broker.topic.load.pending.request.usage")
```

Comment:
Good point. Yes.
@codelipenghui WDYT?

Reply:
I think we can't change it to lookup because multiple topic lookup request can only load one topic. 

Actually, I think it should be 

```
pulsar.broker.topic.concurrent.load.usage
pulsar.broker.topic.concurrent.load.limit
```
We also use `maxConcurrentTopicLoadRequest` in the broker.conf. Keep the name consistent will be better for users. Do not need to translate from concurrent -> pending.

Reply:
@codelipenghui Couple of questions here:
1. Namespace matters. We try to avoid writing in "English": Topic concurrent load --> `pulsar.broker.topic.concurrent.load"...
   Namespace (prefix) means we might have other metrics under it. I don't see any other metrics for `topic.concurrent`. Hence to me it makes more sense to have `pulsar.broker.topic.load.concurrent` and then `usage` and `limit`, where usage is the current number of concurrent load operations, and limit is the defined limit.
2. Regarding the word `operation` suggested. Let's discuss this.
    Lookup is a request right? It's coming from the user in `/lookup` or in the `lookup` command in the binary protocol. Yet, @dragosvictor mentioned that lookup is actually called from within other requests like get topic of namespaces, get partition metadata, etc. Perhaps the correct thing here as @dragosvictor said is that lookup we measure is actually an operation. 
    Same with lopic load. It is not happening due to a request, but an internal decision, hence it is an operation.
    So perhaps the names should actually be:
   - `pulsar.broker.topic.operations.lookup.concurrent.usage`
   - `pulsar.broker.topic.operations.lookup.concurrent.limit`
   - `pulsar.broker.topic.operations.load.concurrent.usage`
   - `pulsar.broker.topic.operations.load.concurrent.usage`
   
@lhotari @codelipenghui Would love your opinion about this

Reply:
@asafm 

Oh, I thought the namespace means Pulsar namespace before.

1. So I understand correctly, we will have a couple of metric namespaces:

- pulsar.broker.*
- pulsar.tenant.*
- pulsar.namespace.*
- pulsar.topic.*
- pulsar.subscription.*

Is it correct? And, `pulsar.broker.topic.load.concurrent.*` also looks good to me.

2. But I think it should be covered by metrics for other APIs? For example, deleting a topic will validate the ownership of the topic. The the requested broker is not the owner of the topic, the broker will return 307 to the HTTP client. I think we should have metrics for topic deletion API. For topic lookup, we usually pay more attention to the client lookup requests and the metadata operations from the lookup request (We will skip the metadata operation if the broker cached metadata), not the internal calls to the broker cached metadata (validate topic ownership). And for the metadata operation, we will have metadata driver level metrics.

And, at least we need `lookup.request` for the client side lookup requests. Otherwise, we don't know how many lookup `operation` is from client side lookup requests or internal topic ownership validation.

Reply:
perhaps generic parts of the metric name could be removed? just wondering if parts like `concurrent` or `operations` are required at all?

Reply:
See response at https://github.com/apache/pulsar/pull/22058#discussion_r1511099159